lib/core/shell/src/lib/shell.routes.ts

```diff
@@ -15,13 +15,52 @@
  * limitations under the License.
  */
 
-import { Route } from '@angular/router';
+import { provideRouter, Route, Routes } from '@angular/router';
 import { ShellLayoutComponent } from './components/shell/shell.component';
 import { SHELL_AUTH_TOKEN } from './services/shell-app.service';
+import { EnvironmentProviders } from '@angular/core';
 
 export const SHELL_LAYOUT_ROUTE: Route = {
     path: '',
     component: ShellLayoutComponent,
     canActivate: [SHELL_AUTH_TOKEN],
     children: []
 };
+
+export interface AppShellRoutesConfig {
+    shellParentRoute?: Route;
+    shellChildren: Routes;
+}
+
+/**
+ * Provides shell routes for the application.
+ *
+ * @param routes The routes configuration for the shell.
+ * @returns An array of providers for the shell routes.
+ */
+export function provideShellRoutes(routes: Routes | AppShellRoutesConfig): EnvironmentProviders[] {
```

Comment:
Can we update documentation and mention about new way and deprecation of ShellModule? There is documentation for that:
lib/core/shell/README.md